app/components/work_packages/activities_tab/journals/new_component.html.erb

```diff
@@ -53,28 +53,41 @@
           },
           url: work_package_activities_path(work_package_id: work_package.id)
         ) do |f|
-          flex_layout(justify_content: :space_between, align_items: :flex_end) do |form_container|
-            form_container.with_column(
-              classes: "work-packages-activities-tab-journals-new-component--ck-editor-column",
-              mr: 2
-            ) do
-              render(WorkPackages::ActivitiesTab::Journals::NotesForm.new(f))
-            end
-            form_container.with_column do
-              render(
-                Primer::Beta::IconButton.new(
-                  scheme: :default,
-                  icon: :"paper-airplane",
-                  "aria-label": t("activities.work_packages.activity_tab.label_submit_comment"),
-                  type: :submit,
-                  data: {
-                    test_selector: "op-submit-work-package-journal-form",
-                    "work-packages--activities-tab--index-target": "formSubmitButton"
-                  }
+          if adding_restricted_comment_allowed?
+            concat(
+              flex_layout do |flex|
+                flex.with_row(mb: 3, test_selector: "op-work-package-journal-restricted-comment-checkbox") do
+                  render(WorkPackages::ActivitiesTab::Journals::RestrictedNoteForm.new(f))
```

Comment:
Clicking on the checkbox closes the comment field unless the user inserts text first. So if the user has the flow of first selecting the visibility, this is annoying.

![Kapture 2025-03-13 at 10 30 36](https://github.com/user-attachments/assets/c7bde95f-8efc-4bef-b05c-310cf4ef0f95)

Reply:
This is a known issue. We are making design changes to the whole comment compose input as follow ups.